src/System.Management.Automation/engine/parser/Parser.cs

```diff
@@ -6426,16 +6426,126 @@ internal Ast CommandRule(bool forDynamicKeyword)
 
         #region Expressions
 
-        private ExpressionAst ExpressionRule()
+        private ExpressionAst ExpressionRule(bool endNumberOnTernaryOpChars = false)
         {
             // G  expression:
             // G      logical-expression
             // G
             // G  logical-expression:
+            // G      binary-expression
+            // G      ternary-expression
+            // G
+            // G  ternary-expression:
+            // G      binary-expression  new-lines:opt   '?'   new-lines:opt   ternary-expression   new-lines:opt   ':'   new-lines:opt   ternary-expression
+
+            RuntimeHelpers.EnsureSufficientExecutionStack();
+            var oldTokenizerMode = _tokenizer.Mode;
+            try
+            {
+                SetTokenizerMode(TokenizerMode.Expression);
+
+                List<Ast> componentAsts = new List<Ast>();
+                ExpressionAst condition = BinaryExpressionRule(endNumberOnTernaryOpChars);
+                if (condition == null)
```

Comment:
Did you consider treating `?` as a binary operator with higher precedence so some of this code can live in the operator precedence parser?

The continuation is an issue of course, but you could optionally enable continuations for all operators.

Reply:
I didn't consider that. On a quick thought, it would require `?` and `:` with the lowest precedence order, and the logic change would make the operator precedence parser more confusing to read.
I don't see the benefit of doing so. `?:` is not a binary operator and IMO, dealing with it in a separate method is cleaner.